Tests/kaas/k8s-version-recency/k8s-version-recency-check.py

```diff
@@ -0,0 +1,465 @@
+                return gt and self.upper_version > version_info
+        else:
+            return self.lower_version == version_info
+
+
+def diff_months(date1, date2):
+    return abs((date1.year - date2.year) * 12 + date1.month - date2.month)
+
+
+def diff_weeks(date1, date2):
+    day1 = (date1 - datetime.timedelta(days=date1.weekday()))
+    day2 = (date2 - datetime.timedelta(days=date2.weekday()))
+    diff = day2 - day1
+    return abs((diff.days / 7) + math.ceil(diff.seconds / 86400))
+
+
+def diff_days(date1, date2):
+    delta = date1 - date2
+    return abs(delta.days)
+
+
+async def request_cve_data(session: aiohttp.ClientSession, cveid: str) -> dict:
+    """Request for a single CVE data item."""
+    resp = await session.request('GET', f"https://cveawg.mitre.org/api/cve/{cveid}",
+                                 headers={"Accept": "application/json"})
+    return await resp.json()
+
+
+def parse_cve_version_information(cve_version_info):
+    """Parse the CVE version information according to their CVE JSON 5.0 schema"""
+    vi_lower_version = None
+    vi_upper_version = None
+    less_than = False
+    equal = False
+
+    # Extract the version if it is viable, but it's not a requirement
+    try:
+        vi_lower_version = K8sVersionInfo().extract_version(cve_version_info['version'].strip("v"))
+    except ValueError:
+        pass
+
+    if 'lessThanOrEqual' in cve_version_info:
+        vi_upper_version = K8sVersionInfo().extract_version(cve_version_info['lessThanOrEqual'].strip("v"))
+        vi_upper_version.patch += 1
+        less_than = True
+        equal = True
+    elif 'lessThan' in cve_version_info:
+        vi_upper_version = K8sVersionInfo().extract_version(cve_version_info['lessThan'].strip("v"))
+        less_than = True
+
+    # This shouldn't happen, but if it happens, we look for non-standard descriptions
+    # According to this(https://www.cve.org/AllResources/CveServices#cve-json-5),
+    # this isn't how the data should be described
+    if not vi_lower_version and not vi_upper_version:
+        if re.search(r'v?\d+.\d+.x', cve_version_info['version']):
+            vdata = cve_version_info['version'].strip("v").split(".")
+            vi_lower_version = K8sVersionInfo(vdata[0], vdata[1], 0)
+            vi_upper_version = K8sVersionInfo(vdata[0], int(vdata[1])+1, 0)
+
+        if re.search(r'v?\d+.\d+.\d+\s+-\s+v?\d+.\d+.\d+', cve_version_info['version']):
+            vdata = cve_version_info['version'].split("-")
+            vi_lower_version = K8sVersionInfo().extract_version(vdata[0].strip("v"))
+            vi_upper_version = K8sVersionInfo().extract_version(vdata[1].strip("v"))
+
+    return CVEVersionInfo(vi_lower_version, vi_upper_version, less_than, equal)
+
+
+async def collect_cve_versions():
+    """Get all relevant CVE versions, that are relevant for the test according to the severity
+    dictated by the Standard.
+    """
+
+    # CVE fix versions
+    cfvs = list()
```

Comment:
Just a suggestion, but wouldn't it make here more sense to have a set instead of a list? So you don't have to check for duplicate entries, and the lookup is faster?

Reply:
I had some problems with a set here, since the `CVEVersionInfo` class is unhashable. I could probably make it hashable, but using a list and comparing seemed like the easier solution to me.